src/cocoa/toga_cocoa/widgets/webview.py

```diff
@@ -71,18 +74,32 @@ async def evaluate_javascript(self, javascript):
         """
         Evaluate a JavaScript expression.
 
-        **This method is asynchronous**. It will return when the expression
+        **This method is asynchronous**. It will return when the expression has been
+        evaluated and a result is available.
 
         :param javascript: The javascript expression to evaluate
         :type  javascript: ``str``
         """
-        return self.native.evaluateJavaScript(javascript, completionHandler=None)
+
+        class JSResult:
+            result = None
+            result_semaphore = Semaphore(0)
```

Comment:
Semaphore is the asyncio primitive; Future is the high-level object. What you've defines as JSResult is effectively an asyncio.Future; you get an instance of the object, set the result in the completion handler, and "await future" to get the return value.

Reply:
Ah, a way to make my life difficult.